### DISCUSSION TITLE
Shouldn't we pass 'conf' in ConfusionMatrix?

### DISCUSSION THREAD

**Discussion by p0p4k** [ultralytics/yolov5#9563] [Q&A] [answered]
Hello, should we pass `conf_thres` for `conf` in the following line? 
`confusion_matrix = ConfusionMatrix(nc=nc, conf=conf_thres) `

https://github.com/ultralytics/yolov5/blob/30fa9b610a3a6d9dc6a9e5961388710e5af0b704/val.py#L186

https://github.com/ultralytics/yolov5/blob/30fa9b610a3a6d9dc6a9e5961388710e5af0b704/utils/metrics.py#L126-L132

**Comment by glenn-jocher** [comment, marked as answer]
No, the problem is mAP should be evaluated at near zero confidence threshold, but the confusion matrix should be produced at a deployment confidence level, ie 0.25 normally.

**Reply by p0p4k** [reply]
Alright, I get it now. Thanks for the swift reply.
